### ISSUE TITLE
bug: attention_based evicts every new token before it can acquire an attention score

### ISSUE THREAD

**Issue by umran666** [umran666/UniqKache#60]
### Bug and location

New metadata starts with zero cumulative attention in [src/uniqkache/cache/metadata.py:128-136](https://github.com/umran666/UniqKache/blob/0b878a1acc3f2a84ea0ff16194456b8ed10191e8/src/uniqkache/cache/metadata.py#L128-L136). [src/uniqkache/cache/kv_cache.py:173-175](https://github.com/umran666/UniqKache/blob/0b878a1acc3f2a84ea0ff16194456b8ed10191e8/src/uniqkache/cache/kv_cache.py#L173-L175) scores and evicts on append, before [src/uniqkache/models/synthetic.py:280-296](https://github.com/umran666/UniqKache/blob/0b878a1acc3f2a84ea0ff16194456b8ed10191e8/src/uniqkache/models/synthetic.py#L280-L296) computes attention and before the caller records the resulting weights.

Once a full `attention_based` cache contains positive-scored tokens, each newly appended token has score zero and is immediately removed. It never participates as a key and can never accumulate evidence that would allow it to survive.

### Reproduction

```python
import torch
from uniqkache import KVCache, build_policy
from uniqkache.models.synthetic import build_model
from uniqkache.metrics.quality import random_token_ids

torch.set_num_threads(1)
model = build_model("tiny")
cache = KVCache(model.config.cache_config(capacity=4),
                policy=build_policy("attention_based"))
prompt = random_token_ids(512, 16)
with torch.no_grad():
    for pos in range(16):
        _, weights = model.forward(prompt[:, pos:pos+1], cache,
                                   start_pos=pos, return_attention=True)
        for layer, weight in enumerate(weights):
            cache.note_attention(layer, weight)
        cache.advance()
print([layer.metadata.positions.tolist() for layer in cache.store.layers])
```

Observed: every layer retains exactly `[0, 1, 2, 3]` after processing position 15. Attention is explicitly recorded and the clock advances in this reproduction.

### Impact and expected behavior

The heavy-hitter policy freezes its early resident set and cannot retain later relevant context. This invalidates comparisons intended to evaluate attention-driven token selection.

Allow newly computed keys to participate in attention and receive a score before deciding whether to evict them. Test admission beyond capacity with real normalized attention, rather than only injecting precomputed scores into policy tests.

### Verification

Reviewed commit: `0b878a1acc3f2a84ea0ff16194456b8ed10191e8`. Reproduced on Windows with Python 3.10 and PyTorch 2.6.0+cu124; CPU unless otherwise specified. Synthetic-model results below are correctness diagnostics, not language-model quality claims.
